### DISCUSSION TITLE
Managing dockerized apps from Sunshine

### DISCUSSION THREAD

**Discussion by Murazaki** [games-on-whales/gow#55] [Q&A]
Hi,

Great effort and stunning results ! Thanks for that.

I would like to know if there's already any documented way to manage dockerized apps launch and stop from sunshine ?
I assume it might mean sunshine needs access to docker.sock to run docker commands ?

(also if you have any idea on how to handle multiple connected players on one session that would be great ?)

**Comment by ABeltramo** [comment]
Thanks for checking out GOW, I'm glad you like it!

> I would like to know if there's already any documented way to manage dockerized apps launch and stop from sunshine ?
I assume it might mean sunshine needs access to docker.sock to run docker commands ?

Me and @zb140 discussed this at the very beginning of this project.   

The _"easy"_ path would be something along the lines of what you suggested: pass the docker socket, install the Docker CLI inside the Sunshine image and let it be able to spin up and down containers as needed.   
I think this can work, but it'll require users to specify a full working `docker start ...` command into the sunshine `apps.json` file with all the proper `volumes`, `env`, etc etc.

This can probably be done easily and pull requests are very welcome! 😉 

We were trying to achieve something a little bit more flexible (https://github.com/games-on-whales/gow/pull/33): a GUI that will run inside a container that can possibly spin up and down other containers by clicking a button. This way we could probably add a little bit of logic behind each container and possibly avoiding users having to manually mess with `docker-compose.yml` or `apps.json`.  

One solution doesn't exclude the other though.

> (also if you have any idea on how to handle multiple connected players on one session that would be great ?)

This is a recurring question on the Discord channel and it's probably worth to turn into a proper issue. 

There are two main use cases for multiple users:
1. Share one game between multiple players for a co-op session
2. Share one machine HW between multiple players with different games running

I think the first one is supported by Sunshine, I haven't tried it but it might be just a matter of setting `channels = 2` in [sunshine.conf](https://github.com/loki-47-6F-64/sunshine/blob/e4c9c292e57d39136df2d46d1e9b66eba53f3bd3/assets/sunshine.conf#L114)

For the second one there are different ways that can possibly achieve that, we still have to try out and see.  
The possible issues that I can see are:
  - How to properly separate different users inputs between Sunshine sessions? 
    - Should we spin up multiple Sunshine + Xorg + PulseAudio instances? They'll still share the same udev, will that cause conflicts?
    - @diadatp suggested a possible alternative approach using [Xorg multiseat](https://wiki.archlinux.org/title/Xorg_multiseat)
